pkg/controller/sparkapplication/controller.go

```diff
@@ -691,20 +695,30 @@ func (c *Controller) submitSparkApplication(app *v1beta2.SparkApplication) *v1be
 				SubmissionAttempts: app.Status.SubmissionAttempts + 1,
 			}
 		}
+
 		c.recordSparkApplicationEvent(app)
 		return app
 	}
 
 	glog.Infof("SparkApplication %s/%s has been submitted", app.Namespace, app.Name)
+	var appState v1beta2.ApplicationStateType
+	if app.Spec.Mode == v1beta2.ClientMode {
```

Comment:
Can you explain this? Why is this state different?

Reply:
submission refers to the spark application and getting resources. In cluster mode pending submission, the spark submit pod is being created and has not succeeded in getting the resources or everything it needs to bring up driver pod. Since we are bypassing the spark submit if the spark application submission succeeds we would automatically bring u the driver pod which means we're in submitted.